File: lc-blind75/ContainsDuplicate(lc-217)/Better/ContainDuplicate.py
def findDuplicate(nums):
    n=len(nums)
    hset=set()
    for i in range(n):
        if nums[i] in hset:
            return True
        hset.add(nums[i])
    return False

n=int(input())
nums=list(map(int,input().split()))
print(findDuplicate(nums))

File: lc-blind75/ContainsDuplicate(lc-217)/Better/ContainDuplicate.java
import java.util.*;
public class ContainDuplicate {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int nums[]=new int[n];
        for(int i=0;i<n;i++){
            nums[i]=sc.nextInt();
        }
        sc.close();
        System.out.println(findDuplicate(nums));
    }
    public static boolean findDuplicate(int nums[]){
        Set<Integer> set=new HashSet<>();
        int n=nums.length;
        for(int i=0;i<n;i++){
            if(set.contains(nums[i])){
                return true;
            }
            set.add(nums[i]);
        }
        return false;
    }
}
